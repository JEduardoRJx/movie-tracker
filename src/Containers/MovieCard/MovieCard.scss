.movie-card {
  display: flex;
  flex-direction: column;
  height: 85vh;
  width: auto;
}

.movie-details {
  height: 100px;
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  color: white;
}

.fav-btn {
  font-size: 1.25em;
  border: none;
  width: 200px;
  border-radius: 5px;
  background-color: lightgray;
  &:hover {
    background-color: #BC8F8F;
  }
}

.poster {
  height: 70vh;
  object-fit: contain;
  &:hover {
    transition: transform .5s ease;
    transform: scale(1.1);
    transform-origin: top;
  }
}