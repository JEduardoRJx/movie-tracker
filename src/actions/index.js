export const addMovies = movies => ({
	type: 'ADD_MOVIES',
	movies
});

export const throwError = errorMsg => ({
  type: 'THROW_ERROR',
  errorMsg
})

export const setUser = user => ({
	type: 'SET_USER',
	user
})

export const toggleFavorite = id => ({
	type: 'FAV_MOVIE',
	id
})

export const getFavs = favs => ({
  type: 'GET_FAVS',
  favs
})
