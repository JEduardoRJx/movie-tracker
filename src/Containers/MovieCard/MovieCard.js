import React from 'react';
import  './MovieCard.scss';
import { toggleFavorite } from '../../actions'
import { connect } from 'react-redux'

const MovieCard = ({ id, title, date, poster, vote, overview, isFavorite, toggleFav }) => {
  const fixedDate = new Date(date + 'T00:00').toString().split(' ').slice(1, 4).join(' ');
  return (
    <div className='movie-card' id={id}>
      <h3>{title}</h3>
      <h3>{fixedDate}</h3>
      <img className='poster' src={`http://image.tmdb.org/t/p/w1280${poster}`} alt='movie poster'/>
      <h3>{vote}</h3>
      <p>{overview}</p>
      <button type="button"
      onClick={() => toggleFav(id) }>Add Favorite</button>
    </div>
  )
}

const mapDispatchToProps = dispatch => ({
  toggleFav: (id) => dispatch( toggleFavorite(id))
})

export default connect(null, mapDispatchToProps)(MovieCard);