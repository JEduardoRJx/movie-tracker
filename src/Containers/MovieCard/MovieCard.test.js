import React from 'react';
import { shallow } from 'enzyme';
import { MovieCard, mapStateToProps, mapDispatchToProps } from './MovieCard';
import { toggleFav } from '../../actions/index';

describe('MovieCard', () => {
  let wrapper, favoriteWrapper;

  it('should match snapshot ', () => {
    favoriteWrapper = shallow (<MovieCard 
      id= '354920'
      title= "Han Solo"
      date='2019/10/28'
      poster={`http://image.tmdb.org/t/p/w1280`} 
      movie
      isFavorite
      toggleFav
      user
      refreshFavs
    />)
    // Expectation
    expect(wrapper).toMatchSnapshot()
  });
  
  it('should call toggleFav when isFavorite is true', () => {

    const mockEvent = { preventDefault: jest.fn() }

    favoriteWrapper.find('button').simulate('click', mockEvent)

    expect(toggleFav).toHaveBeenCalled();
  });
  
})