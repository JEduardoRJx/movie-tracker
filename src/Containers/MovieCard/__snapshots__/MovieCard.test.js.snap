// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard should match snapshot  1`] = `
<article
  className="movie-card"
  id="354920"
>
  <div
    className="movie-details"
  >
    <h2>
      Han Solo
    </h2>
    <h3>
      Date
    </h3>
    <button
      className="fav-btn favorited"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Remove Favorite
    </button>
  </div>
  <Link
    to="/movies/354920"
  >
    <img
      alt="Movie poster"
      className="poster"
      src="http://image.tmdb.org/t/p/w1280http://image.tmdb.org/t/p/w1280"
    />
  </Link>
</article>
`;
